src/cpp/benders/benders_core/BendersBase.cpp

```diff
@@ -520,4 +522,52 @@ std::string BendersBase::status_from_criterion() const
 	default:
 		return Output::STATUS_ERROR_C;
 	}
+}
+
+/*!
+ *  \brief Get path to slave problem mps file from options
+ */
+std::string BendersBase::get_slave_path(std::string const &slave_name) const
+{
+	return (Path(_options.INPUTROOT) / (slave_name + ".mps")).get_str();
+}
+
+/*!
+ *  \brief Return slave weight value
+ *
+ *  \param nslaves : total number of slaves
+ *
+ *  \param name : slave name
+ */
+double BendersBase::slave_weight(int nslaves, std::string const &name) const
+{
+	if (_options.SLAVE_WEIGHT == SLAVE_WEIGHT_UNIFORM)
+	{
+		return 1 / static_cast<double>(nslaves);
+	}
+	else if (_options.SLAVE_WEIGHT == SLAVE_WEIGHT_CONSTANT)
+	{
+		double const weight(_options.SLAVE_WEIGHT_VALUE);
+		return 1 / weight;
+	}
+	else
+	{
+		return _options.weights().find(name)->second;
+	}
+}
```

Comment:
This function is implemented twice

Reply:
yes this is quite redundant. options treatment re-organization is needed.